packages/eslint-plugin/tests/rules/no-invalid-void-type.test.ts

```diff
@@ -98,6 +98,28 @@ function foo(): void | never {
   ],
 });
 
+ruleTester.run('invalidUnionParameter:true', rule, {
+  valid: [
+    {
+      code: 'type allowUnion = string | number',
```

Comment:
@kmin-jeong this PR has some failures. GitHub shows some of them in the web UI (example: the one right here!). You can find the rest of them in the checks view on the main PR page:

<img width="977" alt="image" src="https://user-images.githubusercontent.com/3335181/185404865-ef16fcd0-00fa-4dd2-840e-076fb317ac8d.png">

In particular, we can see tests failing: https://github.com/typescript-eslint/typescript-eslint/runs/7897893084?check_suite_focus=true

You can run just the tests for the `no-invalid-void-type` rule by `cd`ing into `packages/eslint-plugin` and running `npx jest no-invalid-void-type`:

```shell
cd packages/eslint-plugin
npx jest no-invalid-void-type
```